### PR TITLE
Update to Tensorflow version 2.4, Download models in evaluate.py instead of eval_main.py

### PR DESCRIPTION
**Pull request by markus-hinsche**
**Motivation:** Tensorflow 2.4 has great new features for mixed precision, distributed model training, and more (We ran Tensorflow version 2.3 before)

**Changes:**
- upgrade to TF 2.4 (also numpy, and smaller python dependencies)
- upgrade AzureML Docker image used during training
- pin pip version to avoid problems with the new [resolver](https://stackoverflow.com/questions/65122957/resolving-new-pip-backtracking-runtime-issue)
- Removed unused `papermill`,  `nbconvert`
- download model inside `evaluate.py` instead of in `eval_main.py`
- Fix: on windows, git clone is impossible because of filename including colon: `src/common/data_utilities/tests/pickle_files/scans/c571de02-a723-11eb-8845-bb6589a1fbe8/102/pc_c571de02-a723-11eb-8845-bb6589a1fbe8_2021-04-22 13:34:33.302557_102_3.p`

Check:
- training still works, see [this experiment](https://ml.azure.com/runs/training-junkyard-speedup_1625043124_09cc6886?wsid=/subscriptions/9b5bbfae-d5d1-4aae-a2ca-75159c0c887d/resourcegroups/cgm-ml-prod-we-rg/workspaces/cgm-ml-prod-we-azml&tid=3a27c573-ec1a-4734-9cd3-3208af51794b)
- Evaluation pipeline should run, see [this PR](https://github.com/Welthungerhilfe/cgm-ml/pull/389)

### CODE REVIEW

**Review by lvonasek — APPROVED** [review]
I can confirm that I can checkout mhinsche-update-to-tf24-pipeline branch. Main branch I cannot checkout.

File: src/common/evaluation/QA/eval_depthmap_models/eval_configs.sh [hunk 1]
```diff
@@ -2,17 +2,15 @@
...
-python eval_main.py --qa_config_module qa_config_height_clean
+
 python eval_main.py --qa_config_module qa_config_weight
 python eval_main.py --qa_config_module qa_config_weight_dropout
 
 # Combine results
 python combine_results.py --model_measurement height
-
-# Find inaccurate results for different models
-python inaccurate_scans.py
```
Comment (markus-hinsche) [inline src/common/evaluation/QA/eval_depthmap_models/eval_configs.sh:18]: To fix this, I created a follow-up issue: https://github.com/Welthungerhilfe/cgm-ml/issues/397

**Comment by markus-hinsche** [comment]
@gangooteli Ready for review

File: src/models/CNNDepthMap/CNNDepthMap-height/q3-depthmap-plaincnn-height/train_notebook.ipynb [hunk 2]
```diff
@@ -209,15 +209,15 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "curated_env_name = \"cgm-env\"\n",
+    "curated_env_name = \"cgm-v30\"\n",
```
Comment (gangooteli) [inline src/models/CNNDepthMap/CNNDepthMap-height/q3-depthmap-plaincnn-height/train_notebook.ipynb:212]: Just wondering, Should you use same env either v30 or v31?

In File src/common/model_utils/environment.py, you have already cgm_environment(workspace, curated_env_name, env_exist).

Can we not use the same method to get the environment?

**Review by gangooteli — APPROVED** [review]
LGTM.

Added comments which you can take a look. Otherwise no issues in the code as such.

File: src/models/CNNDepthMap/CNNDepthMap-height/q3-depthmap-plaincnn-height/train_notebook.ipynb (hunk 2 above)
Comment (markus-hinsche) [reply src/models/CNNDepthMap/CNNDepthMap-height/q3-depthmap-plaincnn-height/train_notebook.ipynb:212]: This was tackled in https://github.com/Welthungerhilfe/cgm-ml/pull/381

**Pull request merged**